### PR TITLE
Deprecate global element types

### PR DESCRIPTION
**Pull request by ilyachur**
I deprecated global element type variables from nGraph code.
It is needed to avoid issues with tests initialisations when we build inference engine as a static library. For more information have a look this PR: https://github.com/openvinotoolkit/openvino/pull/3219

I already tried to remove these global variables in PR: https://github.com/openvinotoolkit/openvino/pull/3279 but it is not backward compatible changes.

So in this PR I marked as deprecated global element type variables and fixed nGraph build in order to avoid deprecation warnings.

**Majority of changes in this PR are changing `ngraph::element::[f32,i32,u8, ...]` to `ngraph::element::Type_t::[f32,i32,u8, ...]`**

### CODE REVIEW

File: ngraph/core/include/ngraph/type/element_type.hpp [hunk 1]
```diff
@@ -130,21 +131,37 @@ namespace ngraph
...
         extern NGRAPH_API const Type i64;
+        NGRAPH_DEPRECATED("This global element type was deprecated. Please do not use it.")
         extern NGRAPH_API const Type u1;
+        NGRAPH_DEPRECATED("This global element type was deprecated. Please do not use it.")
         extern NGRAPH_API const Type u8;
+        NGRAPH_DEPRECATED("This global element type was deprecated. Please do not use it.")
         extern NGRAPH_API const Type u16;
+        NGRAPH_DEPRECATED("This global element type was deprecated. Please do not use it.")
         extern NGRAPH_API const Type u32;
+        NGRAPH_DEPRECATED("This global element type was deprecated. Please do not use it.")
```
Comment (ilya-lavrenov) [inline ngraph/core/include/ngraph/type/element_type.hpp:164]: let's write what to use instead

**Review by ilya-lavrenov — APPROVED** [review]

File: ngraph/core/include/ngraph/type/element_type.hpp (hunk 1 above)
Comment (ilyachur) [reply ngraph/core/include/ngraph/type/element_type.hpp:164]: Ok, will update the message.

File: ngraph/frontend/onnx_import/include/onnx_import/core/value_info.hpp [hunk 2]
```diff
@@ -75,7 +75,7 @@ namespace ngraph
 
             const std::string& get_name() const { return m_value_info_proto->name(); }
             const PartialShape& get_shape() const { return m_partial_shape; }
-            const element::Type& get_element_type() const
+            const element::Type get_element_type() const
```
Comment (tomdol) [inline ngraph/frontend/onnx_import/include/onnx_import/core/value_info.hpp:78]: When returning by value it's recommeded to drop the const.

```suggestion
            element::Type get_element_type() const
```

File: ngraph/frontend/onnx_import/include/onnx_import/core/tensor.hpp [hunk 3]
```diff
@@ -531,7 +531,7 @@ namespace ngraph
                 return static_cast<Type>(m_tensor_proto->data_type());
             }
 
-            const element::Type& get_ng_type() const
+            const element::Type get_ng_type() const
```
Comment (tomdol) [inline ngraph/frontend/onnx_import/include/onnx_import/core/tensor.hpp:534]: Here I also recommend to return without the `const` specifier or change this method to return the `Type_t` enum instead.

**Review by tomdol — CHANGES_REQUESTED** [review]

File: ngraph/frontend/onnx_import/include/onnx_import/core/value_info.hpp (hunk 2 above)
Comment (ilyachur) [reply ngraph/frontend/onnx_import/include/onnx_import/core/value_info.hpp]: Fixed

File: ngraph/frontend/onnx_import/include/onnx_import/core/tensor.hpp (hunk 3 above)
Comment (ilyachur) [reply ngraph/frontend/onnx_import/include/onnx_import/core/tensor.hpp]: Fixed

**Review by tomdol — APPROVED** [review]

File: ngraph/core/include/ngraph/type/element_type.hpp [hunk 4]
```diff
@@ -91,11 +91,12 @@ namespace ngraph
             size_t bitwidth() const;
             // The name of this type, the enum name of this type
             const std::string& get_type_name() const;
+            bool operator==(const Type_t& other) const;
```
Comment (pelszkow) [inline ngraph/core/include/ngraph/type/element_type.hpp:94]: Have you try to remove all of comparing functions and let allow compiler to use cast to enum (l:127) to do comparators.

File: ngraph/core/include/ngraph/type/element_type.hpp [hunk 5]
```diff
@@ -130,21 +131,50 @@ namespace ngraph
 
         typedef std::vector<Type> TypeVector;
 
+        NGRAPH_DEPRECATED(
+            "This global element type was deprecated. Please use Type_t::undefined instead.")
```
Comment (pelszkow) [inline ngraph/core/include/ngraph/type/element_type.hpp:135]: Just for my curiosity. Why we don't use `constexpr` and use `extern` for variables which might be inline?

Comment (ilyachur) [reply ngraph/core/include/ngraph/type/element_type.hpp:135]: I will recheck, if it works I will change it and remove deprecation.

**Pull request merged**